js/src/behaviors/tabs.ts

```diff
@@ -0,0 +1,129 @@
+ * @param container - Container of tab control component
+ */
+/**
+ * Tabs behaviors
+ */
+export function initTabs() {
+	initTabNavClickListeners();
+	initTabControlClickListeners();
+
+	// update when tabs into the view
+	const tabContainers = Array.from(document.querySelectorAll('tab-container')) as HTMLElement[];
+	for (const tc of tabContainers) {
+		updateTabNav(tc);
+	}
+}
+
+function initTabNavClickListeners() {
+	window.addEventListener('click', (event: Event) => {
+		const target =
+			event.target instanceof Element &&
+			(event.target.closest('[data-tab-nav]') as HTMLButtonElement);
+
+		if (!target) {
+			return;
+		}
+
+		event.preventDefault();
+		const container = event.target.closest('tab-container') as HTMLElement;
+		const tabs = Array.from(container.querySelectorAll('.tab-control')) as HTMLButtonElement[];
+
+		// get current activated tab
+		const currentActiveButton = container.querySelector(
+			'[aria-selected="true"]'
+		) as HTMLButtonElement;
+		const index = parseInt(currentActiveButton?.dataset.tabControl as string);
+
+		// update tab
+		if (index > 1 && target.dataset.tabNav === 'previous') {
+			updateTabState(container, currentActiveButton, index - 1);
+			updateTabPanel(container, index, index - 1);
+		} else if (index < tabs.length && target.dataset.tabNav === 'next') {
+			updateTabState(container, currentActiveButton, index + 1);
+			updateTabPanel(container, index, index + 1);
+		}
+		updateTabNav(container);
+	});
+}
+
+function initTabControlClickListeners() {
+	window.addEventListener('click', (event: Event) => {
+		const target =
+			event.target instanceof Element &&
+			(event.target.closest('.tab-control') as HTMLButtonElement);
+
+		if (!target) {
+			return;
+		}
+
+		event.preventDefault();
+		const container = event.target.closest('tab-container') as HTMLElement;
+		updateTabNav(container);
+	});
+	window.addEventListener('keydown', (event: KeyboardEvent) => {
+		if (event.key !== 'ArrowLeft' && event.key !== 'ArrowRight') {
+			return;
+		}
+
+		const target =
+			event.target instanceof Element &&
+			(event.target.closest('.tab-control') as HTMLButtonElement);
+
+		if (!target) {
+			return;
+		}
+
+		event.preventDefault();
+		const container = event.target.closest('tab-container') as HTMLElement;
+		updateTabNav(container);
+	});
+}
+
+function updateTabNav(container: HTMLElement = document.body) {
+	const tabs = Array.from(container.querySelectorAll('.tab-control')) as HTMLButtonElement[];
+	const previousButton = container.querySelector('.tab-previous') as HTMLButtonElement;
+	const nextButton = container.querySelector('.tab-next') as HTMLButtonElement;
+	const currentActiveButton = container.querySelector(
+		'[aria-selected="true"]'
+	) as HTMLButtonElement;
+	const index = parseInt(currentActiveButton?.dataset.tabControl as string);
+
+	// slide active button into view
+	currentActiveButton.scrollIntoView({ behavior: 'auto', block: 'nearest', inline: 'center' });
+
+	if (index <= 1) {
```

Comment:
I could probably be convinced otherwise, but I don't like how the selection cycles from end back to beginning. @BenDMyers @ollips @jdanyow any thoughts on this?

Reply:
fyi, this is the default of the tab-container functionality.

Reply:
Since that is part of the `tab-container-element`, I am having a hard time intercept the listener to force a stop

Reply:
In that case, let's go ahead and leave in the looping.